File: all-submissions/0002-add-two-numbers/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode head = new ListNode(0);
        ListNode curr = head;
        int carry = 0;
        while(l1 != null && l2 != null){
            int total = l1.val + l2.val + carry;
            int sum = total%10;
            ListNode temp = new ListNode(sum);
            curr.next = temp;
            carry = total/10;
            l1 = l1.next;
            l2 = l2.next;
            curr = curr.next;
            
        }
        while(l1 != null) {
            int total = l1.val+carry;
            int sum = total%10;
            ListNode temp = new ListNode(sum);
            curr.next = temp;
            carry = total/10;
            l1 = l1.next;
            curr = curr.next;
        }

        while(l2 != null) {
            int total = l2.val + carry;
            int sum = total%10;
            ListNode temp = new ListNode(sum);
            curr.next = temp;
            carry = total/10;
            l2 = l2.next;
            curr = curr.next;
        }
        if(carry!=0) {
            curr.next = new ListNode(carry);
        }
        return head.next;
    }
}

File: all-submissions/0002-add-two-numbers/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        dummy_head = ListNode(0)
        temp = dummy_head
        carry = 0
        sum =0

        while (l1 != None and l2 != None):
            sum = l1.val + l2.val + carry
            carry = sum //10
            temp.next =  ListNode(sum %10)
            temp = temp.next
            l1 = l1.next
            l2 = l2.next
        while (l1 != None):
            sum = l1.val +carry
            carry = sum //10
            temp.next =  ListNode(sum %10)
            temp = temp.next
            l1 = l1.next
        while (l2 != None):
            sum = l2.val +carry
            carry = sum //10
            temp.next =  ListNode(sum %10)
            temp = temp.next
            l2 = l2.next
        if carry != 0:
            temp.next =  ListNode(carry)
            temp = temp.next

        return dummy_head.next 
		
		

